### ISSUE TITLE
Automate link checking for generated documentation

### ISSUE THREAD

**Issue by scottfrederick** [spring-projects/spring-boot#21035]
There have been a few issues lately with broken links in docs (e.g. #20818, #21019). We should investigate automated checking of links in generated or published docs.

**Comment by dreis2211** [comment]
I used http://validator.w3.org/checklink today to find hopefully one last issue. Maybe you can take a look at that for inspiration and/or validation of results.

As you can see there are some URLs which are generally reachable via Browser, but not if you simply curl against them, e.g.:
```
curl -I https://redis.io
HTTP/1.1 404 Not Found
Server: nginx/1.10.2
Date: Wed, 22 Apr 2020 16:50:24 GMT
Content-Length: 3673
Connection: keep-alive
```

**Comment by wilkinsona** [comment]
Interesting finding. Thanks, @dreis2211. Looks like redis.io changes behaviour based on the `User-Agent`. It responds with a 200 If you spoof it with curl and pretend to be a browser.

**Comment by dreis2211** [comment]
Yeah, there are some which are also behind a HTTP Basic auth. My point being: Some URLs are false positives and require a bit more magic than others for an eventual tooling.

**Comment by danielmenezesbr** [comment]
Istio.io [[1]](https://github.com/istio/istio.io/blob/7782eb1d0a0d980743bb5c23c63ab3e070749899/scripts/lint_site.sh#L176) [[2]](https://github.com/istio/istio.io/blob/7782eb1d0a0d980743bb5c23c63ab3e070749899/scripts/lint_site.sh#L168) uses [html-proofer](https://github.com/gjtorikian/html-proofer) and [linkinator](https://github.com/JustinBeckwith/linkinator) to test rendered HTML files to make sure they're accurate.
